crates/core/src/server/http_gateway/v1.rs

```diff
@@ -57,8 +61,12 @@ async fn web_home(
         .build();
 
     let token_header = headers::Authorization::bearer(token.as_str()).unwrap();
-    let contract_idx = path_handlers::contract_home(key, rs, token).await?;
-    let mut response = contract_idx.into_response();
+    let contract_response = path_handlers::contract_home(key, rs, token.clone()).await?;
+
+    // Note: We can't store the token in attested_contracts here because we don't have the ContractInstanceId
```

Comment:
mmm, if the current fix works is ok, but I think we should be able to provide the instance somehow from here? is not the `key` ?